qemu/tests/x86_cpu_flag_smap.py

```diff
@@ -0,0 +1,74 @@
+import logging
+
+from avocado.utils import process
+
+from virttest import env_process
+from virttest import utils_misc
+from virttest import error_context
+from provider.cpu_utils import check_cpu_flags
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Test cpu flag smap.
+    1) Check if flag smap is in the supported lists, if no, cancel test
+    2) For q35 machine type
+      2.1) Boot guest with supported cpu model
+      2.2) Check smap flag inside guest(only for linux guest)
+    3) For pc machine types
+      3.1) Check pc machine types that are newer than rhel7.0.0
+      3.2) Boot with new machine type
+      3.3) Check cpu flags inside guest(only for linux guest)
+      3.4) Check old pc machine types that are older than rhel7.0.0
+      3.5) Boot with old machine type
+      3.6) Check cpu flags inside guest(only for linux guest)
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    def get_machine_type(machine_types):
+        """
+        Get pc machine_types supported by current qemu
+        :param machine_types: all machine types
+        :return:older machine types list
+        """
+        pc_old = []
+        for line in machine_types:
+            line = line.split(" ")
+            if line[0].startswith('rhel'):
+                pc_old.append(line[0])
+        return pc_old
+
+    flags = params["flags"]
+    check_host_flags = params.get_boolean("check_host_flags")
+    if check_host_flags:
+        check_cpu_flags(params, flags, test)
+
+    boot_times = 1
+    machine_type = params['machine_type']
+    if machine_type == 'pc':
+        qemu_binary = utils_misc.get_qemu_binary(params)
+        query_cmd = "%s -M ?" % qemu_binary
+        qemu_binary_output = process.system_output(
+            query_cmd, shell=True).decode().splitlines()
+        pc_machine_types_old = get_machine_type(
+            qemu_binary_output)
+        if pc_machine_types_old:
+            boot_times = 2
+    for x in range(boot_times):
+        if x >= 1:
+            flags = ''
+            params['no_flags'] = 'smap'
+            params['machine_type'] = pc_machine_types_old[0]
```

Comment:
You don't have the variant `pc_machine_types_old` when `params['machine_type']` is not `pc`, so this line you might hit error, right?

Reply:
> You don't have the variant `pc_machine_types_old` when `params['machine_type']` is not `pc`, so this line you might hit error, right?

If there is no `pc_machine_types_old` list the VM will only boot once. If there is `pc_machine_types_old` the VM will boot twice, the first time the machine type is pc and the second time the machine type is the latest in the `pc_machine_types_old` list.

Reply:
@nanliu-r Okay, I see. Thanks for clarification.